patterns/design-patterns-2/src/main/java/com/baeldung/prototype/Circle.java

```diff
@@ -0,0 +1,23 @@
+package com.baeldung.prototype;
+
+import java.util.List;
+
+public class Circle extends Shape {
+    private int radius;
+    private List<String> metadata;
+
+    public Circle(int radius, List<String> metadata) {
+        this.radius = radius;
+        this.metadata = metadata;
+    }
+
+    public Circle(Circle that) {
+        this.radius = that.radius;
+        this.metadata = that.metadata;
```

Comment:
This is not a clone operation... it links the internal state of the new `Circle` to the old one. They literally _share_ the `metadata` so modifying it on one circle would affect its clone. You need to deep-copy every property, not just the references.

Obviously this doesn't apply to `int` ;)